src/build_system.ml

```diff
@@ -398,7 +399,15 @@ let compile_rule t ~all_targets_by_dir ?(allow_override=false) pre_rule =
     in
     let targets_min_ts = min_timestamp t targets_as_list  in
     let deps_max_ts    = max_timestamp t all_deps_as_list in
-    if rule_changed ||
+    let is_runtest ps  =
+      let is_runtest p =
+        match Path.extract_build_context p with
+        | Some (".aliases", _) -> true
```

Comment:
Not sure I'm following with the utility function, so I'd just like to clarify:

```ocaml
match Utils.alias_of_file path with
| Some (path_to_alias, basename) -> ...
| None -> ...
```

...where `path_to_alias` is the file path as if it were in the .aliases build context and the string argument represents the basename?